src/rpc.ts

```diff
@@ -99,12 +98,13 @@ export class RPC extends EventEmitter {
     private readonly target: IPostable,
     protocolVersion: string,
     private readonly origin: string = '*',
-    isApp?: boolean,
+    private readonly isApp?: boolean,
     private readonly setOnMessage?: (fn: (ev: any) => void) => void,
     private readonly removeOnMessage?: (fn: (ev: any) => void) => void,
```

Comment:
What we should do instead of these flags is something like 

```ts
interface IReceiveable {
    addListener(listener: (ev: MessageEvent) => any);
    removeListener(listener: (ev: MessageEvent) => any);
}
```

Then make a wrapper that's used depending on the environment, both for IPostable and IReceivable

```ts
const desktopPostable: IPostable = window; // window works fine alone

class DesktopReceivable implements IReceiveable {
    private listener: (ev: any) => void;
    protected readonly handler = (ev: MessageEvent) => this.listener(ev.data);

    constructor(private target: any) {}

    public setListener(listener: (ev: any) => void) {
        this.listener = listener;
        this.target.addEventListener('message', this.handler);
    }

    public removeListener() {
        this.target.removeEventListener('message', this.handler);
    }
}

// document, or whichever, just JSON.stringify before it goes out :)
const appPostable: IPostable = {
    postMessage: (data, origin) => document.postMessage(JSON.stringify(data), origin)
};
class AppReceivable extends DesktopReceivable implements IReceiveable {
    protected readonly handler = (ev: MessageEvent) => this.listener(JSON.parse(ev.data));

  constructor() {
    super(document); // again if you need to
  }
}
```

It's more code but is much more explicit and less fiddly than three different 'flags'. Helps keep logic separated, the RPC can worry about transmitting and reordering packets without caring about how exactly they're sent over the transport.